boards/common/arduino-atmega/include/board_common.h

```diff
@@ -105,6 +105,16 @@ extern "C" {
 #define XTIMER_BACKOFF              (40)
 /** @} */
 
+/**
+ * @name    ztimer configuration values
+ * @{
+ */
+#define CONFIG_ZTIMER_USEC_TYPE     ZTIMER_TYPE_PERIPH
+#define CONFIG_ZTIMER_USEC_DEV      (TIMER_DEV(0))
+#define CONFIG_ZTIMER_USEC_FREQ     (250000LU)
+#define CONFIG_ZTIMER_USEC_WIDTH    (16)
+/** @} */
+
```

Comment:
With changing the frequency selecting to work for different core clocks, this could be provided for all ATmega based boards. So maybe factor this out to e.g. `boards/common/atmega/include/board_atmega_common.h`?

This would yield the 250kHz on 16MHz ATmegas (e.g. most Arduinos), but also work on other ATmegas:
```C
#if CLOCK_CORECLOCK > 4000000UL
#define CONFIG_ZTIMER_USEC_FREQ     (CLOCK_CORECLOCK / 64)
#else
#define CONFIG_ZTIMER_USEC_FREQ     (CLOCK_CORECLOCK / 8)
#endif
```

Reply:
I'm not sure this is needed. The frequency (`250000LU`) is passed as is to periph's `timer_init()`. Why calculate the result if we know it?

But you're right in that probably the config can be moved to `boards/common/atmega/include/board_atmega_common.h` as is! I'll try.

Reply:
On ATmega `timer_init()` fails if it cannot device the core clock down to the specified clock frequency. Some ATmega boards use the integrated 8 MHz oscillator (e.g. one old Arduino type I fail to remeber or the ATmega on a breadboard "boards") or use crystal with a frequency different to 16 MHz (e.g. the waspmote). By calculating the timer clock based on the core clock, the resulting frequency is guaranteed to be usuable on any board.

E.g. the ATmega328P by default runs of the internal 8 MHz clock divided down to 1 MHz. Most (all?) ATmegas can run at most at 20 MHz. The above calculation generates IMO reasonable timer frequencies for the whole range from 1 MHz to 20 MHz.

(Maybe it would be a good idea to also document this reasoning, in case we can agree on this.)

Reply:
I'm just realizing that there is no `boards/common/arduino-atmega/include/board_common.h`.
So I'd like to postpone this to a later PR.
As is, ztimer will re-use xtimer's config, if one is available, so this would be an optimization anyways.